aws_xray_sdk/core/models/entity.py

```diff
@@ -140,7 +141,12 @@ def put_annotation(self, key, value):
             log.warning("ignoring unsupported annotation value type %s.", type(value))
             return
 
-        self.annotations[key] = value
+        sanitized_key = ''.join([c for c in key if c in _valid_annotation_key_characters])
```

Comment:
Annotation key  is difference to entity name. User can have 100 lines of code adding annotations to a single entity. Under worst case this could result in 100 times of string concatenation, which adds a lot of overhead. 

Invalid key is a user error and the key should be dropped until the author explicit fix it using valid characters.